src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/VilkårperiodeService.kt

```diff
@@ -134,15 +143,74 @@ class VilkårperiodeService(
         val behandling = behandlingService.hentSaksbehandling(eksisterendeVilkårperiode.behandlingId)
         validerBehandlingIdErLik(vilkårperiode.behandlingId, eksisterendeVilkårperiode.behandlingId)
         validerBehandling(behandling)
-
-        feilHvis(eksisterendeVilkårperiode.kildeId != vilkårperiode.kildeId) {
-            "Kan ikke oppdatere kildeId på en allerede eksisterende vilkårperiode"
-        }
+        validerKildeIdOgType(vilkårperiode, eksisterendeVilkårperiode)
 
         val grunnlagsdata =
             grunnlagsdataService
                 .hentGrunnlagsdata(behandling.id)
 
+        if (behandling.type != BehandlingType.REVURDERING) {
+            return oppdaterVilkårperiodeHvorAltKanEndres(eksisterendeVilkårperiode, vilkårperiode, behandling, grunnlagsdata)
+        }
+
+        return oppdaterVilkårperiodeIRevurdering(vilkårperiode, eksisterendeVilkårperiode, behandling, grunnlagsdata)
+    }
+
+    @Transactional
+    fun oppdaterVilkårperiodeIRevurdering(
+        vilkårperiode: LagreVilkårperiode,
+        eksisterendeVilkårperiode: Vilkårperiode,
+        behandling: Saksbehandling,
+        grunnlagsdata: Grunnlagsdata,
+    ): Vilkårperiode {
+        val revurderFra = behandling.revurderFra
+        validerRevurderFraErSatt(revurderFra)
+        validerAtVilkårperiodeKanOppdateresIRevurdering(eksisterendeVilkårperiode, revurderFra)
+
+        // Håndter vilkårsperioder hvor alle felter kan oppdateres
+        if (eksisterendeVilkårperiode.fom >= revurderFra) {
+            feilHvis(vilkårperiode.fom < revurderFra) {
+                "Kan ikke sette fom før revurder-fra(${revurderFra.norskFormat()})"
+            }
+            return oppdaterVilkårperiodeHvorAltKanEndres(eksisterendeVilkårperiode, vilkårperiode, behandling, grunnlagsdata)
+        }
+
+        return oppdaterVilkårperiodeFørRevurderFra(vilkårperiode, eksisterendeVilkårperiode, revurderFra, grunnlagsdata)
+    }
+
+    /**
+     * Håndterer alle perioder som enten er helt før eller krysser revurder fra datoen, dvs. perioder hvor kun tom kan endres.
+     * Dersom perioden inneholder vurderinger med svar GAMMEL_MANGLER_DATA får man kun lov å korte ned tom.
+     */
+    @Transactional
+    fun oppdaterVilkårperiodeFørRevurderFra(
+        vilkårperiode: LagreVilkårperiode,
+        eksisterendeVilkårperiode: Vilkårperiode,
+        revurderFra: LocalDate,
+        grunnlagsdata: Grunnlagsdata,
+    ): Vilkårperiode {
+        val vurderingerMedGammelManglerData = eksisterendeVilkårperiode.faktaOgVurdering.vurderinger.inneholderGammelManglerData()
+        val finnesGammelDataUtenomAldersVilkår = vurderingerMedGammelManglerData.filterNot { it is VurderingAldersVilkår }.isNotEmpty()
```

Comment:
```suggestion
vurderingerMedGammelManglerData.any { it !is VurderingAldersVilkår }
```
Litt mange negeringer for mitt hode 🤯

Reply:
Will fix